packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -135,3 +145,33 @@ export function useShowMoversGestures( { ref, debounceTimeout = 500 } ) {
 		},
 	};
 }
+
+/**
+ * Hook that toggles the highlight (outline) state of a block
+ *
+ * @param {string} clientId The block's clientId
+ *
+ * @return {Function} Callback function to toggle highlight state.
+ */
+export function useToggleBlockHighlight( clientId ) {
+	const { toggleBlockHighlight } = useDispatch( 'core/block-editor' );
+
+	const updateBlockHighlight = useCallback(
+		( isFocused ) => {
+			toggleBlockHighlight( clientId, isFocused );
+		},
+		[ clientId ]
+	);
+
+	useEffect( () => {
+		return () => {
+			// Sequences state change to enable editor updates (e.g. cursor
+			// position) to render correctly.
+			requestAnimationFrame( () => {
```

Comment:
I don't have any other suggestions, but this code gives me vibes that it could be fragile (race conditions).

(`requestAnimationFrame` and `setTimeout` are often misused to address issues where we _should_ be acting as a side effect of some specific state transitions, and only be lucky circumstance can the same effect be achieved with scheduled callbacks)

Reply:
This was my only work around :(. I'm not comfortable with the fragile vibes either. There's something happening (elsewhere in the editor) that does something with focus (or perhaps caret position). Without `rAF` the interaction breaks.

I realized this after the E2E tests kept failing (earlier)

Open to suggestions! 🙏

Reply:
I personally wonder if we need this unmount behavior at all. It could be just a matter of including a reducer logic where the flag should be set to false when we start typing or when the block is unselected. (Unless I'm not understanding the purpose of this hook)

Reply:
I merged but if we find a better solution, let's follow-up on it.